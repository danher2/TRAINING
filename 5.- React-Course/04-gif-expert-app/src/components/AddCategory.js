import React, {useState} from 'react'
import PropTypes from 'prop-types';






export const AddCategory = ({setCategories}) => {

    //estado inicial del input, al escribir en el imput
    const [inputValue, setInputValue] = useState('');
   
    const handledInputChange = (e)=>{
        console.log(e.target.value);
        setInputValue(e.target.value);

        console.log('handleInputChange llamado')
    } 

    //al dar enter
    const handledSubmit = (e)=>{
      e.preventDefault(); // quita el refresh del submit, ya no se hace eso
      console.log('handleSubmit', inputValue);

      if(inputValue.trim().length>2){
      setCategories(cats=>[inputValue, ...cats]);
      setInputValue('');
    }
      // console.log("Submit hecho!!!");
      
    }


    return (
        
        <form onSubmit = {handledSubmit}>
          <p>{inputValue}</p>
        <input
                type='text'
                value={inputValue}
                onChange={handledInputChange}
           
           />

        </form>
       
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  }
