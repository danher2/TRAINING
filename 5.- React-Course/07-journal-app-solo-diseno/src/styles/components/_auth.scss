//asegurarse de importar este archivo scss en styles.scss
.auth__main {
    align-items: center; // centrar todo
    background-color: $primary;//color que tenemos en el settings.css $primary -> es la referencia
    display: flex; // caja centrada
    justify-content: center;
    margin: 0px;
    height: 100vh;
    width: 100vw;
}

.auth__box-container {
    background-color: white;
    box-shadow: 0px 3px $dark-grey;
    border-radius: 2px;
    padding: 20px;
    width: 250px;
}


.auth__title {
    color: darken($color: $primary, $amount: 20); // darken = funcion de sass para modificar colores $color:$primary el color base que ya tenemos y $amount: la modificacion
    margin-bottom: 20px;
}

.auth__input {
    color: $dark-grey;
    border: 0px;
    border-bottom: 1px solid $light-grey;
    font-size: 16px;
    margin-bottom: 10px;
    height: 20px;
    width: 100%;

    transition: border-bottom .3s ease; //animar la transicion
    
    &:focus { // se le dice a sass que es la misma clase pero cuando este con el foco
        border-bottom: 1px solid $primary;
        outline: none; // que no tenga ningun foco cuando esta seleccionado
    }
}

.auth__social-networks {
    align-items: center;
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding-top: 20px;
    padding-bottom: 20px;
    width: 100%;
}


.auth__alert-error {  // en el caso que suceda un error colocaremos el background-color en rojo
    background-color: red;
    border-radius: 5px;
    color: white;
    display: flex;
    justify-content: center;
    margin-bottom: 10px;
    padding: 5px
}