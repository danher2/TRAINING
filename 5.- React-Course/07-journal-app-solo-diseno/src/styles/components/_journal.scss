//asegurar de importarlo en los estilos

.journal__main-content {
    display: flex;
}

.journal__sidebar {
    background-color: $dark-grey;
    color: white;
    display: flex;
    flex-direction: column; //columnas
    height: 100vh;
    padding-left: 10px;
    padding-right: 10px;
    width: 450px;
}

.journal__sidebar-navbar {
    display: flex;
    justify-content: space-between;
}

.journal__sidebar-navbar h3 { // todo lo que sea h3 tendra este estilo
    font-weight: lighter;
}

.journal__new-entry {
    align-items: center;
    cursor: pointer; // para que parezca que se puede dar click enel
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 30px;
    width: 100%;

    transition: color .3s ease; // animacion del color

    &:hover {
        color: darken($color: white, $amount: 20); // cambia de color cuando el cursor esta sobre el
    }
    
}

.journal__entries { //scroll bar mas llamativo
    margin-top: 30px;
    overflow: auto;
 
    &::-webkit-scrollbar {
        -webkit-appearance: none;
        background-color: #ccc;
    }
 
    &::-webkit-scrollbar:vertical {
        width: 8px;
    }
 
    &::-webkit-scrollbar:horizontal {
        height: 10px;
    }
 
    &::-webkit-scrollbar-thumb {
        background-color: #797979;
        border-radius: 20px;
    }
} // lo que sea que se pase lo pueda recorrer con el scroll


.journal__entry {
    background-color: white;
    border-radius: 4px;
    color: $dark-grey;
    display: flex;
    margin-bottom: 10px;
    overflow: hidden;
}

.journal__entry-picture {
    height: 75px;
    width: 75px;
}


.journal__entry-body { // para el cuerpo de cada entry
    padding: 10px;
}

.journal__entry-title { // para el title de cada entry
    font-size: 14px;
    font-weight: bold;
}

.journal__entry-content { //para el contenido del entry
    font-size: 10px;
}

.journal__entry-date-box { // para la fecha de cada entry
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 5px;
}

.journal__entry-date-box span{ // estilo de los span
    font-size: 12px;
}
