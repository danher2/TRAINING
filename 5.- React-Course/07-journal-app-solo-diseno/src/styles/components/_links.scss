//asegurarse que este estilo este importado en styles.scss

.link {
    color: $dark-grey;
    text-decoration: none;

    &:hover {// cuando el cursor este sobre el que aparezca el subrayado
        text-decoration: underline;
    }
}