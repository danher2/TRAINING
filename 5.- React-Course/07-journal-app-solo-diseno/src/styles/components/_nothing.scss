//asegurarse de que este estilo este importado en styles.scss
.nothing__main-content {
    align-items: center;
    background-color: $primary;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100vh;

    p {
        text-align: center;
    }
}

