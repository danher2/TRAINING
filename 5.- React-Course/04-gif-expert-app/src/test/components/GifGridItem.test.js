import React from 'react'
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"



describe('Prueba en <GifGridItem />',()=>{

    const title = 'Titulo';
    const url = 'http://www.google.com';
    const wrapper = shallow(<GifGridItem
        title = {title}
        url = {url}
/>)



    test('debe mostrar el componente corractamente ', () => {
        
        
        expect(wrapper).toMatchSnapshot();

    })


    test('Debe tener un parrafo con el titulo ', () => {
        

        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim() ).toBe(title);


    })
    
    test('Debe obtener la imagen igual  al url y alt de los props', () => {

        const img = wrapper.find('img');
        console.log(img.props().src);
    // console.log(img.props('src'));

    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)

        
    })


    test('debe tener animate__rubberBand ', () => {

        const div =  wrapper.find('div');
        console.log(div.props().className);
        const className = div.props().className;

        expect(className.includes('animate__rubberBand')).toBe(true)


    })
    
    

})