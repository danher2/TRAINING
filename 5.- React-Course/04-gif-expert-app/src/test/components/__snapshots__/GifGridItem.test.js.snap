// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> debe mostrar el componente corractamente  1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Titulo"
    src="http://www.google.com"
  />
  <p>
    Titulo
  </p>
</div>
`;
